app/frontend/helpers/ui_helper.js

```diff
@@ -216,6 +216,66 @@ class UiHelper {
     var textLoadingIndicator = this.getCurrentTextLoadingIndicator(tabIndex);
     textLoadingIndicator.hide();
   }
+
+  getFirstVisibleVerseAnchor() {
+    var firstVisibleVerseAnchor = null;
+    var verseListFrame = app_controller.getCurrentVerseListFrame();
+
+    if (verseListFrame != null) {
+      var verseListFrameRect = verseListFrame[0].getBoundingClientRect();
+
+      var currentNavigationPane = app_controller.navigation_pane.getCurrentNavigationPane()[0];
+      var currentNavigationPaneWidth = currentNavigationPane.offsetWidth;
+
+      var firstElementOffsetX = verseListFrameRect.x + currentNavigationPaneWidth + 15;
+      var firstElementOffsetY = verseListFrameRect.y + 15;
+      
+      var firstElement = document.elementFromPoint(firstElementOffsetX, firstElementOffsetY);
```

Comment:
Maybe declare these variables with const/let as they are only relevant to the current block scope

Reply:
Good point.